test/l2pool2d_test.js

```diff
@@ -0,0 +1,264 @@
+'use strict';
+import {l2Pool2d} from '../src/pool2d.js';
+import {Tensor} from '../src/lib/tensor.js';
+
+import * as utils from './utils.js';
+
+describe('test pool2d', function() {
+  it('l2Pool2d default', function() {
+    const x = new Tensor([1, 1, 4, 4], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]);
+    const windowDimensions = [3, 3];
+    const y = l2Pool2d(x, {windowDimensions});
+    utils.checkShape(y, [1, 1, 2, 2]);
+    utils.checkValue(y, [
+      20.639767440550294,
+      23.302360395462088,
+      31.654383582688826,
+      34.51086785347479,
+    ]);
+  });
+
+  it('l2Pool2d nhwc', function() {
+    const x = new Tensor([1, 4, 4, 1], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]);
+    const windowDimensions = [3, 3];
+    const layout = 'nhwc';
```

Comment:
(ignorable musing - no action expected) 🤔 I've often wondered why pooling doesn't have variations that pool across *batches* too, not only the "spatial" dimensions. e.g. One could generalize pooling consistently like reduction (which can apply from all dimensions to none), and pass windowDimensions of rank 4 with `[3, 1, 3, 3]` to pool along batches too, or just `[3, 1, 1, 1]` for only batches. Now, DML *can* achieve the latter via explicit strides (to map the first dimension to the last dimensions), but there's no way to access it via WebNN, and I've never heard of a need for it anyway. 🤷